python/whylogs/api/writer/whylabs.py

```diff
@@ -606,6 +612,33 @@ def _tag_custom_perf_metrics(self, view: Union[DatasetProfileView, SegmentedData
                         self.tag_custom_performance_column(column_name, default_metric=metric)
         return
 
+    def _get_or_create_transaction_client(self) -> TransactionsApi:
+        self._refresh_client()
+        return TransactionsApi(self._api_client)
+
+    def start_transaction(self, **kwargs) -> None:
+        if self._transaction_id is not None:
+            logger.error("Must end current transaction with commit_transaction() before starting another")
+            return
+
+        if kwargs.get("dataset_id") is not None:
+            self._dataset_id = kwargs.get("dataset_id")
+
+        client: TransactionsApi = self._get_or_create_transaction_client()
+        request = TransactionStartRequest(dataset_id=self._dataset_id)
+        result: LogTransactionMetadata = client.start_transaction(request, **kwargs)
+        self._transaction_id = result["transaction_id"]
+
+    def commit_transaction(self, **kwargs) -> None:
+        if self._transaction_id is None:
+            logger.error("Must call start_transaction() before commit_transaction()")
+            return
+
+        client = self._get_or_create_transaction_client()
+        request = TransactionCommitRequest(verbose=True)
+        client.commit_transaction(self._transaction_id, request, **kwargs)
```

Comment:
Consider returning from the commit? Maybe we get status or a code or error message?

Reply:
Verified if throws on failure. There doesn't seem to be anything interesting to return...